code/game/objects/items/ego_weapons/he.dm

```diff
@@ -516,3 +516,110 @@
 	icon_state = "pleasure"
 	happy = FALSE
 	force = 30
+
+/obj/item/ego_weapon/giant
+	name = "giant"
+	desc = "I'll grind your bones to make my bread!"
+	special = "This weapon deflects ranged attacks while attacking and can block on command. \
+				This weapon has a slow attack speed and deals atrocious damage."
+	icon_state = "giant"
+	force = 28
+	attack_speed = 3
+	damtype = RED_DAMAGE
+	armortype = RED_DAMAGE
+	attack_verb_continuous = list("shoves", "bashes")
+	attack_verb_simple = list("shove", "bash")
+	hitsound = 'sound/weapons/genhit2.ogg'
+	attribute_requirements = list(
+							FORTITUDE_ATTRIBUTE = 40
+							)
+
+//Mostly copied from daredevil
+	var/blocktime = 2 SECONDS
+	var/attacking = FALSE
+	var/block = 0
+	var/block_success
+	var/list/reductions = list(40, 20, 40, 20, 1)
+
+/obj/item/ego_weapon/giant/melee_attack_chain(mob/user, atom/target, params)
+	..()
+	if (!istype(user,/mob/living/carbon/human))
+		return
+	attacking = TRUE
+	addtimer(CALLBACK(src, .proc/drop_stance), 2.5 SECONDS)
+
+/obj/item/ego_weapon/giant/proc/drop_stance()
+	attacking = FALSE
+
+/obj/item/ego_weapon/giant/attack_self(mob/user)
```

Comment:
You should modularize this into a new "Shield" weapon type for easy of use and reduction of copypaste.

Reply:
As of now it is the only one of its kind, let me know if you plan to make another

Reply:
Changes are nearly complete, will commit when they are but I will keep daredevil out for the time being to keep this atomic.